cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/replication/PushStrategy.java

```diff
@@ -206,12 +204,9 @@ private void replicate()
         }
 
         this.state.documentCounter = 0;
-        for (this.state.batchCounter = 1; this.state.batchCounter < this.batchLimitPerRun; this
-                .state.batchCounter++) {
+        while (!this.state.cancel) {
```

Comment:
This change to the `while` loop means that if the replication was cancelled, we'd get the logging after the loop indicating that the pull completed, which could prove misleading when debugging from logs.